### DISCUSSION TITLE
NonZero assignment

### DISCUSSION THREAD

**Discussion by jpulecio** [Katsevich-Lab/sceptre#150] [Q&A] [answered]
We have a CRISPR-activation + 10x transcriptomics experiment where we test the effect of targeting candidate enhancer regions to induce the expression of a set of genes that normally are not expressed in our cells (we intentionally chose genes that are not expressed based on bulk RNA-seq data). Is there a way to include the cells that pass the QC tests although they have undetected expression values for our genes of interest?

**Comment by timothy-barry** [comment]
Hi,

QC will be a bit more complicated in your setting. See [here](https://timothy-barry.github.io/sceptre-book/run-qc.html#sec-qc_cellwise_qc) for a discussion of the cell-wise QC parameters. My recommendation would be to play with the `response_n_umis_range` and `response_n_nonzero_range` parameters. You can set these parameters to a value like `c(0, 0.98)`, which filters out cells with extremely high expression but not extremely low expression. Also, you will need to be careful about [pairwise QC](https://timothy-barry.github.io/sceptre-book/run-qc.html#sec-qc_cellwise_qc). My recommendation would be to initially deactivate pairwise QC by setting `n_nonzero_trt_thresh` and `n_nonzero_cntrl_thresh` to `0`. This will ensure that you do not miss any interesting discoveries in which a gene's expression goes from zero to nonzero after perturbation by CRISPRa. You should feel free to post again if you want some more detailed tips about pairwise QC.

**Reply by jpulecio** [reply]
Thank you for the suggestions! We will try that and update the results in here.

**Reply by jpulecio** [reply]
Hi there! It actually worked better with the regular settings in our hands (it includes too much background noise otherwise). A related question we had is about the p-value in the Power Discovery output table. When we assign multiple pairs to the positive controls it generates a power discovery table without the column "significant", in this way we can only rely on the absolute p-value to dissect the pairs that are significant. Can we do something similar with the treatment (gRNAs)? Would it be correct to select the treatments with a p<0.05 if that was the criteria to analyze the positive control group?

**Reply by timothy-barry** [reply]
Hi,

Just as a point of clarification, by "Power Discovery output table," are you referring to the data frame containing the power check results? I.e.,
```
power_df <- get_result(sceptre_object, "run_power_check")
``` 
Right, this table does not contain a column `significant`, as typically there is only a handful of positive control pairs, and typically these pairs produce fairly small p-values. So assessing "significance" somewhat challenging.

I am not totally sure I understand the second part of the question. Could you explain a bit more what you mean by "Can we do something similar with the treatment (gRNAs)?" Do you mean you want to identify a p-value for each gRNA instead of each gRNA-gene pair?

**Reply by ekatsevi** [reply, marked as answer]
Adding on to Tim's point: Since the perturbation-gene associations among the positive control pairs are not in question, there is usually no need to use a multiple testing procedure to select as subset of these pairs showing evidence of association. However, if this were desired (perhaps if you want to see which of your positive control perturbations worked), there is no reason not to subject the positive control pair p-values to a multiple testing procedure like Bonferroni or BH.

**Reply by jpulecio** [reply]
Hi, thanks for the answers. I realized after I posted it that my question was not 100% clear. What I meant is, in the context of a CRISPRa screen if I want to compare the percentage of efficiency to detect a positive control region vs the treatments regions I need a way to call the significant pairs detected for the positive controls. This is important when you consider that there are technical issues, such as number of gRNAs per targeted region or even gRNA binding efficiency that could affect the outcome in a CRISPRa setting. I did exactly what Eugene suggested and ran a multiple testing procedure for the positive controls, using BH and cutoff <0.1, just as it's done with the discovery pairs.
